GVRf/Extensions/3DCursor/3DCursorLibrary/src/main/java/org/gearvrf/io/cursor3d/MovableBehavior.java

```diff
@@ -112,15 +113,16 @@ void handleClickEvent(CursorEvent event) {
         cursorSceneObject = event.getCursor().getSceneObject();
         prevCursorPosition.set(cursor.getPositionX(), cursor.getPositionY(), cursor
                 .getPositionZ());
-        selected = getOwnerObject();
+        selected = true;
         if (cursor.getCursorType() == CursorType.OBJECT) {
             Vector3f position = new Vector3f(cursor.getPositionX(), cursor.getPositionY(),
                     cursor.getPositionZ());
 
-            selected.getTransform().setPosition(-position.x + selected.getTransform()
-                    .getPositionX(), -position.y + selected.getTransform()
-                    .getPositionY(), -position.z + selected.getTransform().getPositionZ());
-            cursorSceneObject.addChildObject(selected);
+            ownerObject.getTransform().setPosition(-position.x + ownerObject.getTransform()
```

Comment:
Do you need a null check here?

Will this call be made once onDetach() is called in line 225?